webapp/components/DeleteData/DeleteData.vue

```diff
@@ -1,80 +1,46 @@
-              {{
-                $t('settings.deleteUserAccount.commentedCount', {
-                  count: currentUser.commentedCount,
-                })
-              }}
-            </label>
-            <ds-space margin-bottom="small" />
-            <ds-section id="delete-user-account-warning">
-              <div v-html="$t('settings.deleteUserAccount.accountWarning')"></div>
-            </ds-section>
-          </div>
-        </transition>
-      </ds-container>
-      <template slot="footer" class="delete-data-footer">
-        <ds-container>
-          <div
-            class="delete-input-label"
-            v-html="$t('settings.deleteUserAccount.pleaseConfirm', { confirm: currentUser.name })"
-          ></div>
-          <ds-space margin-bottom="xx-small" />
-          <ds-flex :gutter="{ base: 'xx-small', md: 'small', lg: 'large' }">
-            <ds-flex-item :width="{ base: '100%', sm: '100%', md: '100%', lg: 1.75 }">
-              <ds-input v-model="enableDeletionValue" class="enable-deletion-input" />
-            </ds-flex-item>
-            <ds-flex-item :width="{ base: '100%', sm: '100%', md: '100%', lg: 1 }">
-              <base-button
-                icon="trash"
-                danger
-                filled
-                :disabled="!deleteEnabled"
-                data-test="delete-button"
-                @click="handleSubmit"
-              >
-                {{ $t('settings.deleteUserAccount.name') }}
-              </base-button>
-            </ds-flex-item>
-          </ds-flex>
-        </ds-container>
-      </template>
-    </ds-card>
-  </div>
+  <base-card class="delete-data">
+    <h2 class="title">
+      <base-icon name="warning" />
+      {{ $t('settings.deleteUserAccount.name') }}
+    </h2>
+    <label>
+      {{ $t('settings.deleteUserAccount.pleaseConfirm', { confirm: currentUser.name }) }}
+    </label>
+    <ds-input v-model="enableDeletionValue" />
+    <p class="notice">{{ $t('settings.deleteUserAccount.accountDescription') }}</p>
+    <label v-if="currentUser.contributionsCount" class="checkbox">
+      <input type="checkbox" v-model="deleteContributions" />
+      {{
+        $t('settings.deleteUserAccount.contributionsCount', {
+          count: currentUser.contributionsCount,
+        })
+      }}
+    </label>
+    <label v-if="currentUser.commentedCount" class="checkbox">
+      <input type="checkbox" v-model="deleteComments" />
+      {{
+        $t('settings.deleteUserAccount.commentedCount', {
+          count: currentUser.commentedCount,
+        })
+      }}
+    </label>
+    <section v-if="deleteEnabled" class="warning">
+      <p>{{ $t('settings.deleteUserAccount.accountWarning') }}</p>
+    </section>
+    <base-button
+      icon="trash"
+      danger
+      filled
+      :disabled="!deleteEnabled"
+      data-test="delete-button"
+      @click="handleSubmit"
+    >
+      {{ $t('settings.deleteUserAccount.name') }}
+    </base-button>
+  </base-card>
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-12T13:04:48Z" title="Wednesday, February 12th 2020, 2:04:48 pm +01:00">Feb 12, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,80 +1,46 @@
 <template>
-  <div>
-    <ds-card hover>
-      <ds-space />
-      <ds-container>
-        <ds-flex>
-          <ds-flex-item :width="{ base: '22%', sm: '12%', md: '12%', lg: '8%' }">
-            <base-icon name="warning" class="delete-warning-icon" />
-          </ds-flex-item>
-          <ds-flex-item :width="{ base: '78%', sm: '88%', md: '88%', lg: '92%' }">
-            <ds-heading>{{ $t('settings.deleteUserAccount.name') }}</ds-heading>
-          </ds-flex-item>
-          <ds-space />
-          <ds-heading tag="h4">
-            {{ $t('settings.deleteUserAccount.accountDescription') }}
-          </ds-heading>
-        </ds-flex>
-      </ds-container>
-      <ds-space />
-      <ds-container>
-        <transition name="slide-up">
-          <div v-if="deleteEnabled">
-            <label v-if="currentUser.contributionsCount" class="checkbox-container">
-              <input type="checkbox" v-model="deleteContributions" />
-              <span class="checkmark"></span>
-              {{
-                $t('settings.deleteUserAccount.contributionsCount', {
-                  count: currentUser.contributionsCount,
-                })
-              }}
-            </label>
-            <ds-space margin-bottom="small" />
-            <label v-if="currentUser.commentedCount" class="checkbox-container">
-              <input type="checkbox" v-model="deleteComments" />
-              <span class="checkmark"></span>
-              {{
-                $t('settings.deleteUserAccount.commentedCount', {
-                  count: currentUser.commentedCount,
-                })
-              }}
-            </label>
-            <ds-space margin-bottom="small" />
-            <ds-section id="delete-user-account-warning">
-              <div v-html="$t('settings.deleteUserAccount.accountWarning')"></div>
-            </ds-section>
-          </div>
-        </transition>
-      </ds-container>
-      <template slot="footer" class="delete-data-footer">
-        <ds-container>
-          <div
-            class="delete-input-label"
-            v-html="$t('settings.deleteUserAccount.pleaseConfirm', { confirm: currentUser.name })"
-          ></div>
-          <ds-space margin-bottom="xx-small" />
-          <ds-flex :gutter="{ base: 'xx-small', md: 'small', lg: 'large' }">
-            <ds-flex-item :width="{ base: '100%', sm: '100%', md: '100%', lg: 1.75 }">
-              <ds-input v-model="enableDeletionValue" class="enable-deletion-input" />
-            </ds-flex-item>
-            <ds-flex-item :width="{ base: '100%', sm: '100%', md: '100%', lg: 1 }">
-              <base-button
-                icon="trash"
-                danger
-                filled
-                :disabled="!deleteEnabled"
-                data-test="delete-button"
-                @click="handleSubmit"
-              >
-                {{ $t('settings.deleteUserAccount.name') }}
-              </base-button>
-            </ds-flex-item>
-          </ds-flex>
-        </ds-container>
-      </template>
-    </ds-card>
-  </div>
+  <base-card class="delete-data">
+    <h2 class="title">
+      <base-icon name="warning" />
+      {{ $t('settings.deleteUserAccount.name') }}
+    </h2>
+    <label>
+      {{ $t('settings.deleteUserAccount.pleaseConfirm', { confirm: currentUser.name }) }}
+    </label>
+    <ds-input v-model="enableDeletionValue" />
+    <p class="notice">{{ $t('settings.deleteUserAccount.accountDescription') }}</p>
+    <label v-if="currentUser.contributionsCount" class="checkbox">
+      <input type="checkbox" v-model="deleteContributions" />
+      {{
+        $t('settings.deleteUserAccount.contributionsCount', {
+          count: currentUser.contributionsCount,
+        })
+      }}
+    </label>
+    <label v-if="currentUser.commentedCount" class="checkbox">
+      <input type="checkbox" v-model="deleteComments" />
+      {{
+        $t('settings.deleteUserAccount.commentedCount', {
+          count: currentUser.commentedCount,
+        })
+      }}
+    </label>
+    <section v-if="deleteEnabled" class="warning">
+      <p>{{ $t('settings.deleteUserAccount.accountWarning') }}</p>
+    </section>
+    <base-button
+      icon="trash"
+      danger
+      filled
+      :disabled="!deleteEnabled"
+      data-test="delete-button"
+      @click="handleSubmit"
+    >
+      {{ $t('settings.deleteUserAccount.name') }}
+    </base-button>
+  </base-card>
```


I :heart: this refactor! I was just pairing with @roschaefer as he was defining the issue for the `admin` feature to delete user accounts, and we were talking about reusing this component, and what we'd need to change. I was looking at this code that I added and wanting a chance to refactor.
You cut it down from `74` lines of code to `39`, something like that! It so much cleaner as well. I have to admit that I liked using `ds-flex` since it made it kind of easier, but when you know how to achieve the same with `CSS`, it keeps the `html` much cleaner

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2020-02-17T10:01:16Z" title="Monday, February 17th 2020, 11:01:16 am +01:00">Feb 17, 2020</time>_

---


Glad you like it! 💚